### PR TITLE
[Unofficial] VB Prototype "Checked expressions"

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
Unofficial prototype for [VB Lang: Issue 461 Checked Expressions](https://github.com/dotnet/vblang/issues/461)

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Operators.vb [hunk 1]
```diff
@@ -212,6 +212,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 diagnostics = New DiagnosticBag()
             End If
 
+            Dim NoOverflowChecks As Boolean = CheckOverflow
```
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Operators.vb]: Need to rename `NoOverflowChecks` to `HasOverflowChecks` as it better reflects its intention.

File: src/Compilers/VisualBasic/Test/Semantic/Semantics/BinaryOperators.vb [hunk 2]
```diff
@@ -1502,6 +1502,25 @@ End Module
...
+        <Fact>
+        Public Sub UncheckedExpression()
+
+            Dim compilationDef =
+<compilation>
+    <file name="a.vb">
+Module M
+    Sub Main()
+        Dim ten = 10
+        Dim i2 = Unchecked ( 2147483647  + ten )
```
Comment (paul1956) [inline src/Compilers/VisualBasic/Test/Semantic/Semantics/BinaryOperators.vb]: Integer.MaxValue would make intent clearer

Comment (paul1956) [reply src/Compilers/VisualBasic/Test/Semantic/Semantics/BinaryOperators.vb]: The other test that I didn't see is
```
Dim i3 as ULong= UnChecked (ULong.MaxValue + 1)
``` 
which today gives an Error BC30439 Constant expression not representable in type 'ULong'

**Comment by jaredpar** [comment]
We do not accept unexpected language feature pull requests. Please follow the documented policy for adding features to the language

https://github.com/dotnet/roslyn/blob/master/docs/contributing/Developing%20a%20Language%20Feature.md

**Pull request closed without merging**